src/IB/IIMethod.cpp

```diff
@@ -233,6 +233,58 @@ IIMethod::getFEDataManager(const unsigned int part) const
     return d_fe_data_managers[part];
 } // getFEDataManager
 
+NumericVector<double>*
+IIMethod::getMeshCoordinates(bool is_current_configuration, std::string time, unsigned int part)
+{
+    if (is_current_configuration)
+    {
+        NumericVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
+        NumericVector<double>* X_vec = nullptr;
+
+        if (time == "current")
+        {
+            X_vec = d_X_current_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "half")
+        {
+            X_vec = d_X_half_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "new")
+        {
+            X_vec = d_X_new_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "ib_ghost")
+        {
+            // already assigned X_ghost_vec
+        }
```

Comment:
```suggestion
```
This routine should just return the configuration. The question of how it is ghosted should be handled elsewhere.